From 0599a626270e85a7c6291f12642f239a024e1ca4 Mon Sep 17 00:00:00 2001
From: Halla Rempt <halla@valdyas.org>
Date: Fri, 27 Sep 2024 12:59:44 +0200
Subject: [PATCH] [ffmpeg] Fix compilation on MacOS

Meson is quite stupid in understanding what language the library is
being built. And the linker flags are selected in a per-language
basis. On MacOS libavfilter has a few objc files, which makes
meson be silently confused and rejecting all the link flags for
the library (even though it continues to use `cc` for the linker).

Personally, I would prefer Meson to error out on such ambiguities,
it would save me two(!) days of debugging time.
---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index b5cbd9a8b3..18d3a2fd1a 100644
--- a/meson.build
+++ b/meson.build
@@ -2746,6 +2746,7 @@ foreach built_lib : built_libs
     link_with: link_with,
     link_whole: static_lib,
     link_args: link_args,
+    link_language: 'c',
     c_args: c_args,
     vs_module_defs: vs_module_def,
     implicit_include_directories : false,
-- 
2.39.5 (Apple Git-154)

